File: CarlaErik/UvA.SoftCon.Questionnaire/AST/Model/Questionnaire.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using UvA.SoftCon.Questionnaire.AST.Model;
using UvA.SoftCon.Questionnaire.AST.Model.Statements;

namespace UvA.SoftCon.Questionnaire.AST
{
    /// <summary>
    /// Represents the root of a questionnaire AST.
    /// </summary>
    public sealed class Questionnaire : Node
    {
        public IReadOnlyList<IStatement> Statements
        {
            get;
            private set;
        }

        public Questionnaire(IReadOnlyList<IStatement> statements)
        {
            Statements = statements;
        }

        public override void Accept(IASTVisitor visitor)
        {
            visitor.Visit(this);
        }
    }
}

File: Amir/qlProject/src/qlProject/ast/Questionnaire.java
package qlProject.ast;

import java.util.List;

import qlProject.ast.statement.IStatement;
import qlProject.ast.statement.IStatementsVisitor;

public class Questionnaire {

	List<IStatement> statementsList;

	public Questionnaire (List<IStatement> statements){
		statementsList = statements;
	}


	public List<IStatement> getStatements ()
	{
		return statementsList;
	}

	public void accept(IStatementsVisitor visitor){
		visitor.visit(this);
	}
}